azure/datalake/store/lib.py

```diff
@@ -177,21 +199,36 @@ class DatalakeRESTInterface:
         'MKDIRS': ('put', set(), set()),
         'OPEN': ('get', set(), {'offset', 'length', 'read'}),
         'RENAME': ('put', {'destination'}, {'destination'}),
-        'TRUNCATE': ('post', {'newlength'}, {'newlength'}),
         'SETOWNER': ('put', set(), {'owner', 'group'}),
-        'SETPERMISSION': ('put', set(), {'permission'})
+        'SETPERMISSION': ('put', set(), {'permission'}),
+        'SETEXPIRY': ('put', {'expiryOption'}, {'expiryOption', 'expireTime'}),
+        'SETACL': ('put', {'aclSpec'}, {'aclSpec'}),
+        'MODIFYACLENTRIES': ('put', {'aclSpec'}, {'aclSpec'}),
+        'REMOVEACLENTRIES': ('put', {'aclSpec'}, {'aclSpec'}),
+        'REMOVEACL': ('put', set(), set()),
+        'MSGETACLSTATUS': ('get', set(), set()),
+        'REMOVEDEFAULTACL': ('put', set(), set())
     }
 
     def __init__(self, store_name=default_store, token=None,
-                 url_suffix=default_adls_suffix, **kwargs):
+                 url_suffix=default_adls_suffix, api_version='2016-11-01', **kwargs):
         # in the case where an empty string is passed for the url suffix, it must be replaced with the default.
         url_suffix = url_suffix or default_adls_suffix
         self.local = threading.local()
         if token is None:
             token = auth(**kwargs)
         self.token = token
-        self.head = {'Authorization': 'Bearer ' + token['access']}
-        self.url = 'https://%s.%s/webhdfs/v1/' % (store_name, url_suffix)
+
+        # There is a case where the user can opt to exclude an API version, in which case
+        # the service itself decides on the API version to use (it's default).
+        if api_version:
```

Comment:
the "or" operator returns the first value than can be considered true using "bool" method. And there is no such element, returns the latest element.
1 or 2 => 1
0 or 2 => 2
0 or [] or {} or "" or 15 => 15